### PR TITLE
export ObjectStore::set_schema_version()

### PR DESCRIPTION
**Pull request by zaki50**
export `set_schema_version()` for https://github.com/realm/realm-java/issues/3417

### CODE REVIEW

**Comment by zaki50** [comment]
please review this @realm/java 

There is one failing test case. I'm ignore it for now since the case is failing in other PRs (for example in #141 , https://ci.realm.io/job/ObjectStore%20Pull%20Request/314/)

File: src/object_store.hpp [hunk 1]
```diff
@@ -47,6 +47,10 @@ class ObjectStore {
     // get the last set schema version
     static uint64_t get_schema_version(Group const& group);
 
+    // set schema version
+    // FIXME remove this after integrating OS's migration related logic into Realm java
+    static void set_schema_version(Group& group, uint64_t version);
```
Comment (cmelchior) [inline src/object_store.hpp]: Can this throw in some cases? E.g. if the version number if lower than the existing. This should be documented.

**Comment by cmelchior** [comment]
Apart from a little comment 👍

File: src/object_store.hpp (hunk 1 above)
Comment (zaki50) [inline src/object_store.hpp]: added more comment

**Comment by kneth** [comment]
:+1:

**Pull request merged**